frontend/tests/controllers/ProblemUpdateTest.php

```diff
@@ -2506,4 +2506,107 @@ public function testUpdateStatementProblemViaSubmit() {
             $response['statement']['markdown']
         );
     }
+
+    public function testCommitModifiedZipExcludesEasyCases(): void {
+        // Get a problem
+        $problemData = \OmegaUp\Test\Factories\Problem::createProblem();
+        $problem = $problemData['problem'];
+        $identity = $problemData['author'];
+
+        //Get the current ZIP file with all the files
+        $artifacts = new \OmegaUp\ProblemArtifacts(
+            $problem->alias
+        );
+        $currentZipPath = $artifacts->getZip();
+
+        // Define changes
+        $pathsToExclude = [
+            'cases/sample',
+            'cases/easy.01',
+        ];
+        $filesToAdd = [
+            'cases/easy.001.in'  => "10 5\n",
+            'cases/easy.001.out' => "15\n",
+        ];
+        $pathsToRename = [
+            'cases/medium.'  => 'cases/hard.',
+        ];
+
+        $deployer = new \OmegaUp\ProblemDeployer(
+            $problem->alias
+        );
+
+        //  Commit the changes to the ZIP file
+        $deployer->commitModifiedZip(
+            'Test complex edit on cases',
+            $identity,
+            $currentZipPath,
+            $pathsToExclude,
+            $filesToAdd,
+            $pathsToRename,
+        );
+        if (is_file($currentZipPath)) {
+            unlink($currentZipPath);
+        }
+        // Retrieve the modified ZIP file
+        $newArtifacts = new \OmegaUp\ProblemArtifacts(
+            $problem->alias
+        );
+        $newZipPath = $newArtifacts->getZip();
+
+        $zip = new \ZipArchive();
+        $this->assertTrue(
+            $zip->open($newZipPath) === true,
+            'The resulting ZIP file could not be opened.'
+        );
+
+        // Excluded: the new ZIP file must not contain any files beginning with these prefixes.
+        for ($i = 0; $i < $zip->numFiles; $i++) {
+            $name = $zip->getNameIndex($i);
+
+            foreach ($pathsToExclude as $excludePrefix) {
+                if (str_starts_with($name, $excludePrefix)) {
+                    $this->fail("The excluded file {$name} still exists.");
+                }
```

Comment:
Acá podrías usar directamente una aseveración de `PHPUnit`:
```suggestion
                $this->assertStringStartsNotWith(
                    $excludePrefix,
                    $name,
                    "The excluded file {$name} still exists."
                );
```